support/cas-server-support-pac4j-core-clients/src/main/java/org/apereo/cas/support/pac4j/web/flow/DelegatedClientAuthenticationAction.java

```diff
@@ -156,6 +161,10 @@ protected Event doExecute(final RequestContext context) throws Exception {
                     this.authenticationSystemSupport.handleAndFinalizeSingleAuthenticationTransaction(service, clientCredential);
                 final TicketGrantingTicket tgt = this.centralAuthenticationService.createTicketGrantingTicket(authenticationResult);
                 WebUtils.putTicketGrantingTicketInScopes(context, tgt);
+
+                // Prepare for future Single Logout - save the profile
+                prepareForFutureSingleLogout(client, credentials, webContext, tgt.getId());
```

Comment:
Name this method more precisely; It's storing a profile so lets call it that.

Reply:
OK!